sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -531,7 +531,7 @@ object GpuCast extends Arm {
               castStringToFloats(trimmed, ansiMode,
                 GpuColumnVector.getNonNestedRapidsType(toDataType))
             case ByteType | ShortType | IntegerType | LongType =>
-              castStringToInts(trimmed, ansiMode,
+              com.nvidia.spark.rapids.jni.CastStrings.stringToInteger(trimmed, ansiMode,
```

Comment:
The main point of doing a custom cast kernel is to eliminate extra work across multiple kernels in the plugin to achieve the desired result.  If we're simply changing which kernels we're calling but still calling the same number of kernels then I'm not sure that actually makes much of a difference in practice.  Even if we are unable to eliminate a kernel call in the plugin, we should be turning on a cast by default or doing _something_ else besides just calling a different kernel, otherwise one wonders why we cannot continue calling the original cudf kernel since the plugin doesn't seem to care or know the difference?

For example, there's a `.strip()` earlier in this code that should not be necessary.  It was only added because the original cudf kernel didn't handle whitespace properly, so it was a workaround.  A Spark-specific casting kernel shouldn't need any prep work on the input, as ideally it should handle all the input cases directly without pre- or post-workarounds.